### ISSUE TITLE
Graph Build on FsToolkit.ErrorHandling reports projects with no Items.

### ISSUE THREAD

**Issue by baronfel** [ionide/proj-info#109]
**Describe the bug**

When opening [FsToolkit.ErrorHandling](https://github.com/demystifyfp/FsToolkit.ErrorHandling) at commit `f09472a88a6f7032b1f6eaf7961454f91c992870` in Ionide 5.5.3 with `"FSharp.enableMSBuildProjectGraph": true` set, the projects cracked contain no items. The `notifyWorkspace` notifications from FSAC look like:

```json
{
    "Kind": "project",
    "Data": {
        "Project": "/workspaces/FsToolkit.ErrorHandling/tools/FunctionMap/FunctionMap.fsproj",
        "Files": [],
        "Output": "/workspaces/FsToolkit.ErrorHandling/tools/FunctionMap/bin/Debug/net5.0/FunctionMap.dll",
        "ProjectReferences": [
            {
                "RelativePath": "../../src/FsToolkit.ErrorHandling/FsToolkit.ErrorHandling.fsproj",
                "ProjectFileName": "/workspaces/FsToolkit.ErrorHandling/src/FsToolkit.ErrorHandling/FsToolkit.ErrorHandling.fsproj"
            },
            {
                "RelativePath": "../../src/FsToolkit.ErrorHandling.TaskResult/FsToolkit.ErrorHandling.TaskResult.fsproj",
                "ProjectFileName": "/workspaces/FsToolkit.ErrorHandling/src/FsToolkit.ErrorHandling.TaskResult/FsToolkit.ErrorHandling.TaskResult.fsproj"
            },
            {
                "RelativePath": "../../src/FsToolkit.ErrorHandling.JobResult/FsToolkit.ErrorHandling.JobResult.fsproj",
                "ProjectFileName": "/workspaces/FsToolkit.ErrorHandling/src/FsToolkit.ErrorHandling.JobResult/FsToolkit.ErrorHandling.JobResult.fsproj"
            }
        ],
        "PackageReferences": [
            {
                "Name": "FSharp.Core",
                "Version": "5.0.0",
                "FullPath": "/root/.nuget/packages/fsharp.core/5.0.0/lib/netstandard2.0/FSharp.Core.dll"
            },
            {
                "Name": "Hopac",
                "Version": "0.3.23",
                "FullPath": "/root/.nuget/packages/hopac/0.3.23/lib/netstandard1.6/Hopac.Core.dll"
            },
            {
                "Name": "Hopac",
                "Version": "0.3.23",
                "FullPath": "/root/.nuget/packages/hopac/0.3.23/lib/netstandard1.6/Hopac.Platform.dll"
            },
            {
                "Name": "Hopac",
                "Version": "0.3.23",
                "FullPath": "/root/.nuget/packages/hopac/0.3.23/lib/netstandard1.6/Hopac.dll"
            },
            {
                "Name": "Ply",
                "Version": "0.3.1",
                "FullPath": "/root/.nuget/packages/ply/0.3.1/lib/netstandard2.0/Ply.dll"
            }
        ],
        "References": [],
        "OutputType": "exe",
        "Info": {
            "IsTestProject": false,
            "Configuration": "Debug",
            "IsPackable": false,
            "TargetFramework": "net5.0",
            "TargetFrameworkIdentifier": ".NETCoreApp",
            "TargetFrameworkVersion": "v5.0",
            "RestoreSuccess": true,
            "TargetFrameworks": [],
            "RunCmd": {
                "Command": "/workspaces/FsToolkit.ErrorHandling/tools/FunctionMap/bin/Debug/net5.0/FunctionMap",
                "Arguments": ""
            },
            "IsPublishable": true
        },
        "Items": [],
        "AdditionalInfo": {}
    }
}
```

for example. 

**Expected behaviour**
Projects cracked should contain items.

The project _does_ work/build with using `dotnet build -graph` on the command line, and checking the structured build log shows that the SourceFiles are in fact being passed to the `CoreCompile`/`Fsc` targets as expected.

**Comment by TheAngryByrd** [comment]
I looked into this a bit tonight and I'm kind of stumped. When running the build code:  https://github.com/ionide/proj-info/blob/ecb0ed9053a3d1e4e8c510e8e55d17650a5af88b/src/Ionide.ProjInfo/Library.fs#L436-L443, it produces a [GraphBuildResult](https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.graph.graphbuildresult?view=msbuild-16-netcore) with the [OverallResult](https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.graph.graphbuildresult.overallresult?view=msbuild-16-netcore#Microsoft_Build_Graph_GraphBuildResult_OverallResult) of `Failure` and a `null` [Exception](https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.graph.graphbuildresult.exception?view=msbuild-16-netcore).  When digging into the [ResultsByTarget](https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.execution.buildresult.resultsbytarget?view=msbuild-16-netcore#Microsoft_Build_Execution_BuildResult_ResultsByTarget), it shows that `CoreCompile` target is the one failing on all projects. However, I do not encounter this build failure when reverting to `5.0.103`, only on `5.0.201`.  Without a good error message, I'm going to be stuck in the mud here. Maybe @cartermp has some deep lore that can point me in a good direction. 

Currently the only work around I have is either turn off project graph, or use `5.0.103` in your global.json.

**Comment by cartermp** [comment]
Hmm, that sounds like there might be a bug in the SDK or MSBuild, offhandedly. We didn't change anything in the F# compiler AFAIK.

**Comment by TheAngryByrd** [comment]
So I wasn't looking into the `StringWriter` where we were do get better logs and found:

```
Using \"Fsc\" task from assembly \"/usr/local/share/dotnet/sdk/5.0.202/FSharp/FSharp.Build.dll\".
Task \"Fsc\"
The specified task executable location \"/usr/local/share/dotnet/sdk/5.0.202/FSharp/fsc.exe\" is invalid.
Done executing task \"Fsc\" -- FAILED.
Done building target \"CoreCompile\" in project \"FsToolkit.ErrorHandling.fsproj\" -- FAILED.
Done building project \"FsToolkit.ErrorHandling.fsproj\" -- FAILED.
```

It looks like it's trying to use a windows executable on macos. Any pointers?

**Comment by Krzysztof-Cieslak** [comment]
I thought we've already fixed it https://github.com/ionide/proj-info/pull/100

**Comment by TheAngryByrd** [comment]
Switching `CoreCompile` to `Build` here still outputs the same error message.

**Comment by Krzysztof-Cieslak** [comment]
I think the core fix was https://github.com/ionide/proj-info/pull/100/files#diff-44471b59d78aad3cf24d57a7b262f424f22d24d33090ace15bf5ce8345d4a53fR88-R92

Is it also done for GraphLoader?

**Comment by TheAngryByrd** [comment]
Hmm I added this right before it builds but had no difference. 

```fsharp

                match System.Environment.GetEnvironmentVariable "DOTNET_HOST_PATH" with
                | null
                | "" -> System.Environment.SetEnvironmentVariable("DOTNET_HOST_PATH", Ionide.ProjInfo.Paths.dotnetRoot)
                | _alreadySet -> ()
```

Maybe there's some deep property I gotta mangle in [BuildParameters](https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.execution.buildparameters?view=msbuild-16-netcore) or [GraphBuildRequestData](https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.graph.graphbuildrequestdata?view=msbuild-16-netcore)?

**Comment by baronfel** [comment]
closed by #111
